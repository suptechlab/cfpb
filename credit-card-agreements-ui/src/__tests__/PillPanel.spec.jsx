import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { shallow } from 'enzyme';
import ReduxPillPanel, { PillPanel, mapDispatchToProps } from '../PillPanel'
import renderer from 'react-test-renderer'

function setupEnzyme() {
  const props = {
    filters: [{fieldName: 'company', value: 'Acme'}],
    clearAll: jest.fn()
  }

  const target = shallow(<PillPanel {...props} />);

  return {
    props,
    target
  }
}

function setupSnapshot(initialQueryState={}) {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: initialQueryState
  })

  return renderer.create(
    <Provider store={store}>
      <ReduxPillPanel />
    </Provider>
  )
}

describe('component:PillPanel', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot({
      company: ['Apples', 'Bananas are great'],
    });
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('does not render when there are no filters', () => {
    const target = setupSnapshot();
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('allows the user to clear all filters', () => {
    const { target, props } = setupEnzyme()
    const button = target.find('.clear-all button');

    button.simulate('click');
    expect(props.clearAll).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    it('hooks into removeAllFilters', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).clearAll();
      expect(dispatch.mock.calls.length).toEqual(1);
    })
  })
})
