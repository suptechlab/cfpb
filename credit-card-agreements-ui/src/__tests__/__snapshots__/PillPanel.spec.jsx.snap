// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:PillPanel does not render when there are no filters 1`] = `null`;

exports[`component:PillPanel renders without crashing 1`] = `
<section
  className="pill-panel"
>
  <h4
    className="pill-label flex-fixed"
  >
    Filters Applied:
  </h4>
  <ul
    className="layout-row"
  >
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        Apples
      </span>
      <button
        onClick={[Function]}
        title="Remove Apples as a filter"
      >
        <span
          className="cf-icon cf-icon-delete"
        />
      </button>
    </li>
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        Bananas are great
      </span>
      <button
        onClick={[Function]}
        title="Remove Bananas are great as a filter"
      >
        <span
          className="cf-icon cf-icon-delete"
        />
      </button>
    </li>
    <li
      className="clear-all"
    >
      <button
        className="a-btn a-btn__link body-copy"
        onClick={[Function]}
      >
        Clear all filters
      </button>
    </li>
  </ul>
</section>
`;
